AMW_angular/io/src/app/apps/apps.component.ts

```diff
@@ -0,0 +1,173 @@
+import { Component, computed, inject, OnInit, Signal } from '@angular/core';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { LoadingIndicatorComponent } from '../shared/elements/loading-indicator.component';
+import { AsyncPipe } from '@angular/common';
+import { IconComponent } from '../shared/icon/icon.component';
+import { PageComponent } from '../layout/page/page.component';
+import { AppsFilterComponent } from './apps-filter/apps-filter.component';
+import { AuthService } from '../auth/auth.service';
+import { ReleasesService } from '../settings/releases/releases.service';
+import { AppsService } from './apps.service';
+import { Release } from '../settings/releases/release';
+import { takeUntil } from 'rxjs/operators';
+import { ToastService } from '../shared/elements/toast/toast.service';
+import { AppServer } from './app-server';
+import { AppsServersListComponent } from './apps-servers-list/apps-servers-list.component';
+import { PaginationComponent } from '../shared/pagination/pagination.component';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { AppServerAddComponent } from './app-server-add/app-server-add.component';
+import { toSignal } from '@angular/core/rxjs-interop';
+import { AppAddComponent } from './app-add/app-add.component';
+import { ResourceService } from '../resource/resource.service';
+import { Resource } from '../resource/resource';
+import { AppCreate } from './app-create';
+
+@Component({
+  selector: 'app-apps',
+  standalone: true,
+  imports: [
+    AppsFilterComponent,
+    AppsServersListComponent,
+    AsyncPipe,
+    IconComponent,
+    LoadingIndicatorComponent,
+    PageComponent,
+    PaginationComponent,
+  ],
+  templateUrl: './apps.component.html',
+})
+export class AppsComponent implements OnInit {
+  private appsService = inject(AppsService);
+  private authService = inject(AuthService);
+  private modalService = inject(NgbModal);
+  private releaseService = inject(ReleasesService); // getCount -> getReleases(0, count)
+  private resourceService = inject(ResourceService);
+  private toastService = inject(ToastService);
+
+  releases: Signal<Release[]> = toSignal(this.releaseService.getReleases(0, 50), { initialValue: [] as Release[] });
+  appServerGroups = toSignal(this.resourceService.getByType('APPLICATIONSERVER'), {
+    initialValue: [] as Resource[],
+  });
+  appServers = this.appsService.apps;
+  count = this.appsService.count;
+  maxResults = this.appsService.limit;
+  offset = this.appsService.offset;
+  filter = this.appsService.filter;
+  releaseId = this.appsService.releaseId;
+  private error$ = new BehaviorSubject<string>('');
+  private destroy$ = new Subject<void>();
+
+  canCreateApp = false;
+  canCreateAppServer = false;
+  canViewAppList = false;
+  isLoading = false;
+
+  currentPage: number;
+  lastPage: number;
+
+  loadingPermissions = computed(() => {
+    if (this.authService.restrictions().length > 0) {
```

Comment:
refactor to a guard clause

Reply:
Not sure, this is needed to receive the signal update after userpermissions are provided